packages/fx-core/src/plugins/resource/bot/v2/plugin.ts

```diff
@@ -235,59 +263,11 @@ export class TeamsBotV2Impl {
     throw new Error("Invalid programming language");
   }
 
-  private resolveActRoles(ctx: Context, inputs: Inputs): PluginActRoles[] {
-    let actRoles: PluginActRoles[] = [];
-
-    const solutionSettings = ctx.projectSetting.solutionSettings as AzureSolutionSettings;
-    // Null check solutionSettings
-    const capabilities = solutionSettings.capabilities;
-    if (capabilities?.includes(PluginActRoles.Bot)) {
-      const scenarios = inputs?.[AzureSolutionQuestionNames.Scenarios];
-      if (isBotNotificationEnabled() && Array.isArray(scenarios)) {
-        const scenarioActRoles = scenarios
-          .map((item) => QuestionBotScenarioToPluginActRoles.get(item))
-          .filter((item): item is PluginActRoles => item !== undefined);
-        // dedup
-        actRoles = [...new Set([...actRoles, ...scenarioActRoles])];
-      } else {
-        // for legacy bot
-        actRoles.push(PluginActRoles.Bot);
-      }
-    }
-
-    if (capabilities?.includes(PluginActRoles.MessageExtension)) {
-      actRoles.push(PluginActRoles.MessageExtension);
-    }
-
-    return actRoles;
-  }
-
-  private resolveHostType(ctx: Context): HostType {
+  private resolveServiceType(ctx: Context): ServiceType {
```

Comment:
I'd like to rename it as decideServiceType() since it is not simply resolved by file scheme

Reply:
According to the discussion, since it's simply resolved through mapping, will still use `resolveServiceType` here.